bullet/src/FreeGroupFeatures.cc

```diff
@@ -0,0 +1,45 @@
+#include "FreeGroupFeatures.hh"
+
+#include <ignition/math/eigen3/Conversions.hh>
+
+namespace ignition {
+namespace physics {
+namespace bullet {
+
+Identity FreeGroupFeatures::FindFreeGroupForModel(
+    const Identity &_modelID) const
+{
+  const auto &model = this->models.at(_modelID)->model;
+
+  // If there are no links at all in this model, then the FreeGroup functions
+  // will not work properly, so we'll just reject these cases.
+  if (model->getNumLinks() == 0)
+    return this->GenerateInvalidId();
+
+  // Reject also if the model has fixed base
+  if (model->hasFixedBase())
+    return this->GenerateInvalidId();
+
+  return _modelID;
+}
+
+/////////////////////////////////////////////////
+void FreeGroupFeatures::SetFreeGroupWorldPose(
+    const Identity &_groupID,
+    const PoseType &_pose)
+{
+  // Convert pose
+  const auto poseTranslation = _pose.translation();
+  const auto poseLinear = _pose.linear();
+  btTransform baseTransform;
+  baseTransform.setOrigin(convertVec(poseTranslation));
+  baseTransform.setBasis(convertMat(poseLinear));
+
+  // Set base transform
+  const auto &model = this->models.at(_groupID)->model;
```

Comment:
There seems to be something odd here, the function receives a group_ID, but operates on the model IDs, have any clue why is this happening like this?

Reply:
A FreeGroupWorld has a model inside of it and I think this function sets the pose of that model so the Group has that pose

Reply:
Correct me if I'm wrong, the model queried `this->models.at(_groupID)` gets the `models` unordered map from the base class. The code seems to imply with the naming `_groupID` that there is a list of groups, and I think at the moment we don't have groups defined anywhere. 

I think you are saying that a Model can be a FreeWorldGroup?

Reply:
I think it can